### PR TITLE
Delete snapshot API.

### LINKED ISSUE #383: Implement TableAdmin::DeleteSnapshot.
Utility function to delete a snapshot. #329

### PR DESCRIPTION
**Pull request by paprat**
This fixes #383.

### CODE REVIEW

File: bigtable/client/table_admin_test.cc [hunk 1]
```diff
@@ -405,14 +405,14 @@ name: 'projects/the-project/instances/the-instance/tables/the-table'
 }
 
 /**
- * @test Verify that `bigtable::TableAdmin::CreateTable` supports
+ * @test Verify that `bigtable::TableAdmin::DeleteTable` supports
```
Comment (coryan) [inline bigtable/client/table_admin_test.cc]: Ouch, good catch, do you mind moving this to a separate PR?

**Comment by paprat** [comment]
Ping.
